packages/devextreme/js/__internal/grids/grid_core/sticky_columns/utils.ts

```diff
@@ -1,43 +1,110 @@
-import { StickyPosition } from './const';
+import { STICKY_BORDER_WIDTH, StickyPosition } from './const';
 
-const getStickyOffsetCore = function (columns, widths: number[], columnIndex): string {
-  let result = 0;
+export const getColumnFixedPosition = (
+  { fixedPosition }: { fixedPosition: StickyPosition | undefined },
+): StickyPosition => fixedPosition ?? StickyPosition.Left;
+
+const getColumnWidth = (column): number => (column.visibleWidth ?? column.width) as number;
```

Comment:
Question: did you check if the column width can be set by text: "200px" or "20%"?